client/js/app/stores/ExplorerStore.js

```diff
@@ -84,6 +101,99 @@ function _getDefaultFilterCoercionType(explorer, filter) {
 }
 
 /**
+ * Runs through the changeset and moves data around if necessary
+ * @param {Object} explorer   The explorer model that is being updated
+ * @param {Object} updates    The updated explorer model
+ * @return {Object}           The new set of updates
+ */
+function _prepareUpdates(explorer, updates) {
+  var newModel = _.assign({}, explorer, updates);
+
+  newModel = _removeEmailExtractionFields(explorer, newModel);
+  newModel = _migrateToFunnel(explorer, newModel);
+  newModel = _migrateFromFunnel(explorer, newModel);
```

Comment:
This code reads really weird. The fact that we call `migrateToFunnel` then immediately after `migrateFromFunnel` is a confusing read. It might make sense to do the check in this function and then only call one or the other. While three lines in a row like this is nice and clean, having to read those other functions to understand why these 2 lines don't contradict themselves isn't the best.

These function names are nice and clear to me on their own, so leveraging an if statement here and only calling one or the other would make for a great and clear read of what's going on.